src/main/jhipster/src/main/webapp/app/crop-settings-manager/name-types/name-types-pane.component.ts

```diff
@@ -157,17 +157,60 @@ export class NameTypesPaneComponent implements OnInit {
     }
 
     deleteNameType(nameType: any) {
-        const confirmModalRef = this.modalService.open(ModalConfirmComponent as Component);
-        confirmModalRef.componentInstance.title = this.translateService.instant('crop-settings-manager.confirmation.title');
-        confirmModalRef.componentInstance.message = this.translateService.instant('crop-settings-manager.name-type.modal.delete.warning', { param: nameType.name });
-        confirmModalRef.result.then(() => {
-            this.nameTypeService.deleteNameType(nameType.id).toPromise().then((result) => {
-                this.alertService.success('crop-settings-manager.name-type.modal.delete.success');
-                this.loadNameType();
-            }).catch((response) => {
-                this.alertService.error('error.custom', { param: response.error.errors[0].message });
-            });
-        }, () => confirmModalRef.dismiss());
+        this.nameTypeService.getMetadata(nameType.id).toPromise().then((metadata) => {
+            let deleteNameTypeFormStudyPromise = Promise.resolve({});
+            let deleteNameTypeFormGermplasmListPromise = Promise.resolve({});
+
+            if (metadata.germplasm > 0) {
+                this.alertService.error('crop-settings-manager.name-type.validation.delete.error.used.in.germplasm');
+
+            } else {
+                const confirmModalRef = this.modalService.open(ModalConfirmComponent as Component);
+                const messages = [];
+
+                if (metadata.studies > 0) {
+                    messages.push(this.translateService.instant('crop-settings-manager.name-type.modal.delete.summary.study'));
```

Comment:
perhaps deleteNameType can be made `async`, using `await` could reduce nesting in this part (see GermplasmImportReviewComponent.save as an example)

Reply:
I am, agree.